### PR TITLE
Payment details page

### LINKED ISSUE #545: Redesigned payment details
Screenshot 2023-09-20 at 3 15 05 PM

fullscreen, available from the send success view

### PR DESCRIPTION
**Pull request by benalleng**
Closes #545

https://receive-page.mutiny-web.pages.dev/

### CODE REVIEW

File: src/components/ActivityDetail.tsx [hunk 1]
```diff
@@ -0,0 +1,654 @@
...
+                    amount={amount() ?? "0"}
+                    price={state.price}
+                    positive={!isSend()}
+                />
+            </Show>
+        </div>
+    );
+}
+
+export const KeyValue: ParentComponent<{ key: string; bigGap?: boolean }> = (
```
Comment (futurepaul) [inline src/components/ActivityDetail.tsx:161]: we use this in other places as well we should probably move it to `Misc`

File: src/components/ActivityDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,654 @@
...
+) => {
+    return (
+        <li
+            class="flex items-start justify-between"
+            classList={{
+                "gap-6": !props.bigGap,
+                "gap-24": props.bigGap
+            }}
+        >
+            <span class="max-w-[75px] text-sm font-semibold uppercase text-gray-400">
```
Comment (futurepaul) [inline src/components/ActivityDetail.tsx:172]: I know this is a nit but don't want to forget: you want `m-grey-400` here. `gray-400` is a tailwind color and has some blue in it. WAY different lol

**Comment by futurepaul** [comment]
this looks really nice, stoked on this.

I like what you did with the `<Route>` ... I usually use file based routing, but I like this a lot for the case of having and `id` it's more explicit (also we might move off of solid start eventually so this is good for that too)

we should do a call reviewing it, some stuff I hope we can simplify code-wise (I'm always a little suspicious when we have to add more props to an existing layout component, such as ` bigGap?: boolean`, so want to see if we can trim those down a bit)

my big open Q: on wide screens what do we do? probably simplest thing is to make the back button show up there as well?

Screenshot 2023-09-25 at 1 48 46 PM

I know you haven't done visual polish yet just making sure you're aware of these:
Screenshot 2023-09-25 at 2 02 26 PM

**Comment by benalleng** [comment]
> make the back button show up there as well

I added a noHide prop to the backLink, this is the component that seems to have gained the most complexity.

File: src/routes/Receive.tsx [hunk 3]
```diff
@@ -512,22 +503,30 @@ export default function Receive() {
...
-                                    <ExternalLink
-                                        href={mempoolTxUrl(
-                                            paymentTx()?.txid,
-                                            network
-                                        )}
+                                {/*TODO: Confirmation time estimate still not possible needs to be implemented in mutiny-node first*/}
+                                <Show when={receiveState() === "paid"}>
+                                    <p
+                                        class="cursor-pointer underline"
+                                        onClick={() =>
```
Comment (futurepaul) [inline src/routes/Receive.tsx:510]: might be nicer to lift this up above the JSX as a function

File: src/routes/Send.tsx [hunk 4]
```diff
@@ -642,16 +644,27 @@ export default function Send() {
...
-                                            network
-                                        )}
-                                    >
-                                        {i18n.t("common.view_transaction")}
-                                    </ExternalLink>
-                                </Show>
+                                <p
+                                    class="cursor-pointer underline"
+                                    onClick={() =>
+                                        navigate(
```
Comment (futurepaul) [inline src/routes/Send.tsx:650]: also lift this as a function above the jsx?

File: src/routes/Swap.tsx [hunk 5]
```diff
@@ -326,22 +323,24 @@ export default function Swap() {
                                     </div>
                                 </div>
                                 <hr class="w-16 bg-m-grey-400" />
-                                <Show
-                                    when={
-                                        channelOpenResult()?.channel?.outpoint
+                                <p
+                                    class="cursor-pointer underline"
+                                    onClick={() =>
+                                        navigate(
```
Comment (futurepaul) [inline src/routes/Swap.tsx:329]: refactor as fn

File: src/components/ActivityDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,644 @@
...
+    );
+}
+
+export function MiniStringShower(props: { text: string }) {
+    const [copy, copied] = useCopy({ copiedTimeout: 1000 });
+
+    return (
+        <div class="grid w-full grid-cols-[minmax(0,_1fr)_auto] gap-1">
+            <TruncateMiddle text={props.text} />
+            {/* <pre class="truncate text-neutral-300 font-light">{props.text}</pre> */}
```
Comment (futurepaul) [inline src/components/ActivityDetail.tsx:183]: cleanup looks like I left this here in the old detailsmodal

**Review by futurepaul — COMMENTED** [review]
this looks pretty good to me just some tiny cleanup comments. @benthecarman / @TonyGiorgio would love to see if you guys have any input

**Comment by benthecarman** [comment]
I think I liked it as a modal better, but everything else looks great

**Comment by TonyGiorgio** [comment]
For the force close screen, can we add an indication that the funds may take a few days to be swept back into the wallet? That's a very common support request.

**Comment by benalleng** [comment]
image

File: src/routes/Receive.tsx [hunk 7]
```diff
@@ -47,12 +46,7 @@ import {
 import { useI18n } from "~/i18n/context";
 import { Network } from "~/logic/mutinyWalletSetup";
```
Comment (futurepaul) [inline src/routes/Receive.tsx:47]: unused var

**Review by futurepaul — APPROVED** [review]
just this unused var (not sure how that got through the linter) but otherwise I'm inclined to merge

@benthecarman I hear you on the modal... I feel like this style gives us more room / flexibility in the long run, instead of trying to cram details into a modal. but let me know if you seriously hate it and we can discuss

**Comment by benthecarman** [comment]
I feel like we could get the best of both worlds doing like a "full page modal" so it doesn't feel like you've moved to another screen

**Comment by benalleng** [comment]
I'll make a commit that makes this a fullscreen modal in the same style as a comparison

**Review by futurepaul — APPROVED** [review]
just one typescript error and then we're good to go!

File: src/components/ActivityDetailsModal.tsx [hunk 8]
```diff
@@ -0,0 +1,598 @@
...
+
+import bolt from "~/assets/icons/bolt.svg";
+import chain from "~/assets/icons/chain.svg";
+import copyIcon from "~/assets/icons/copy.svg";
+import shuffle from "~/assets/icons/shuffle.svg";
+import {
+    ActivityAmount,
+    AmountFiat,
+    AmountSats,
+    DefaultMain,
```
Comment (futurepaul) [inline src/components/ActivityDetailsModal.tsx:25]: unused var

**Pull request merged**